Scripts/Enemy.cs

```diff
@@ -0,0 +1,125 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.Monetization;
+
+public class Enemy : MonoBehaviour
+{
+    [Header("Enemy Info")]
+    // enemy turns around every now and then - coroutine with timeer
+    public int turnTime = 2;
+    // how quickly can this enemy shoot at the player?
+    public float fireRate = 3f;
+    // can the enemy flip?
+    public bool flip;
+    public GameObject bullet;
+    private GameObject player;
+    public Transform bulletSpawnPoint;
+
+    [Header("Assign in Inspector")]
+    [SerializeField] private ParticleSystem deathParticles;
+    [SerializeField] private GameObject coinPrefab;
+
+    public float reach = 2;
+    public bool inRange = false;
+    public Transform currentTarget;
+    private int i = 0;
+    public bool shootingPlayer;
+
+
+    public float nextFire;
+    private float timer;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        player = GameObject.FindGameObjectWithTag("Player");
+        nextFire = Time.time;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (GameManager.current.gamePaused) return;

```

Comment:
An enemy class shouldn't have to know about GameManager. Should be done the other way around. 

Have a bool in Enemy IsActive, which GameManager sets it to true or false. That boolean is then checked here instead of checking GamePause from GameManager